001-cron.php

```diff
@@ -56,6 +56,21 @@ public function block_admin_page() {
 	}
 }
 
+// Non-public multisites can't be reached by our cron runners, so should use Core's native approach
+if ( is_multisite() ) {
+	$details = get_blog_details( get_current_blog_id(), false );
+
+	// get_blog_details() uses numeric strings for backcompat
+	if ( '1' !== $details->public || in_array( '1', array( $details->archived, $details->spam, $details->deleted ), true ) ) {
+		add_filter( 'wpcom_vip_go_use_core_cron', '__return_true' );
+	}
+}
```

Comment:
Can we just move this whole block into a function instead of the filter-based approach?